src/core/app/authHooks.ts

```diff
@@ -0,0 +1,96 @@
+import { appConfig } from '~core/app_config';
+import { currentUserAtom, eventFeedsAtom, featureFlagsAtom } from '~core/shared_state';
+import { yandexMetrics } from '~core/metrics';
+import { loadFeatures, setFeatures } from './features';
+import type { AuthSuccessResponse } from '~core/auth/client/AuthClient';
+import type { CurrentUser } from './user';
+import type { ApiClient } from '~core/api_client';
+import type { BackendFeed } from '~core/auth/types';
+
+export async function onPublicLogin(apiClient: ApiClient) {
+  // load features for public user, without token
+  const featuresResponse = await loadFeatures(apiClient, false);
+  setFeatures(featuresResponse);
+}
+
+export async function onLogin(apiClient: ApiClient, response: AuthSuccessResponse) {
+  const jwtUserdata = {
+    username: response.jwtData.preferred_username,
+    token: response.token,
+    email: response.jwtData.email,
+    firstName: response.jwtData.given_name,
+    lastName: response.jwtData.family_name,
+    id: response.jwtData.sub,
+  };
+
+  extLoginTasks(jwtUserdata);
```

Comment:
no need to shorten
is that external login tasks?
could've also be extensive, extended

Reply:
totally agree, let's not shorten names